TaskSharper.WPF.Common/Components/Notification/NotificationViewModel.cs

```diff
@@ -130,54 +160,81 @@ private Task ShowNotification(Events.Resources.Notification notification)
+                            if (dateTimeMin.StartsWith("0"))
+                                dateTimeMin = dateTimeMin.TrimStart('0');
+
+                            var textFormat = LocalizeDictionary.Instance
+                                .GetLocalizedObject("NotificationPastEvent", null, LocalizeDictionary.Instance.Culture)
+                                .ToString();
+                            NotificationStart = string.Format(textFormat, NotificationEventType.ToLower(),
+                                dateTimeMin, startToEndTime);
+                        }
+                        else if (DateTime.Now < notification.Event.Start.Value)
+                        {
+                            TimeSpan substractedDateTime = notification.Event.Start.Value.Subtract(DateTime.Now);
+                            dateTimeMin = new DateTime(substractedDateTime.Ticks).AddMinutes(1).ToString("mm");
+                            if (dateTimeMin.StartsWith("0"))
+                                dateTimeMin = dateTimeMin.TrimStart('0');
+
+                            var textFormat = LocalizeDictionary.Instance
+                                .GetLocalizedObject("NotificationPressentEvent", null, LocalizeDictionary.Instance.Culture)
+                                .ToString();
+                            NotificationStart = string.Format(textFormat, NotificationEventType.ToLower(), dateTimeMin, startToEndTime);
+                        }
+                    }
                 }
-                else
-                {
-                    TimeSpan substractedDateTime = notification.Start.Value.Subtract(DateTime.Now);
-                    dateTimeMin = new DateTime(substractedDateTime.Ticks).AddMinutes(1).ToString("mm");
-                    if (dateTimeMin.StartsWith("0"))
-                        dateTimeMin = dateTimeMin.TrimStart('0');
-
-                    var textFormat = LocalizeDictionary.Instance
-                        .GetLocalizedObject("NotificationPastEvent", null, LocalizeDictionary.Instance.Culture)
-                        .ToString();
-                    NotificationStart = string.Format(textFormat, NotificationEventType.ToLower(), dateTimeMin);
-                }   
             }
             IsPopupOpen = true;
             System.Windows.Threading.Dispatcher.CurrentDispatcher.Invoke(() =>
             {
-                var path = "Media/WindowsNotifyCalendar.wav";
+                var path = AppDomain.CurrentDomain.BaseDirectory + "/Media/WindowsNotifyCalendar.wav";
                 SoundPlayer notificationSound = new SoundPlayer(path);
-                notificationSound.Load();
-                notificationSound.Play();
+                try
+                {
+                    notificationSound.Load();
+                    notificationSound.Play();
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e);
+                    throw;
```

Comment:
Handle this exception properly - rethrowing an exception like this is the same as not doing try-catch at all.

Reply:
ops

Reply:
fixed and logging it